superset-frontend/src/dashboard/util/getFilterConfigsFromFormdata.js

```diff
@@ -20,89 +20,85 @@
-      if (config.defaultValue) {
-        defaultValues = config.defaultValue.split(';');
-      }
-      const updatedColumns = {
-        ...columns,
-        [config.column]: config.vals || defaultValues,
-      };
-      const updatedLabels = {
-        ...labels,
-        [config.column]: config.label,
-      };
+  const configs = { columns: {}, labels: {} };
 
-      return {
-        columns: updatedColumns,
-        labels: updatedLabels,
-      };
-    },
-    { columns: {}, labels: {} },
-  );
+  filter_configs.forEach(({ column, label, defaultValue, vals }) => {
+    const defaultValues =
+      typeof defaultValue === 'string' ? defaultValue.split(';') : defaultValue;
+    configs.columns[column] = vals || defaultValues;
+    configs.labels[column] = label;
+  });
 
   if (date_filter) {
-    let updatedColumns = {
-      ...configs.columns,
-      [TIME_FILTER_MAP.time_range]: form_data.time_range,
-    };
-    const updatedLabels = {
-      ...configs.labels,
-      ...Object.entries(TIME_FILTER_MAP).reduce(
-        (map, [key, value]) => ({
-          ...map,
-          [value]: TIME_FILTER_LABELS[key],
-        }),
-        {},
-      ),
-    };
-
+    configs.columns[TIME_FILTER_MAP.time_range] = form_data.time_range;
+    // a map from frontend enum key to backend column
+    Object.entries(TIME_FILTER_MAP).forEach(([key, column]) => {
+      configs.labels[column] = TIME_FILTER_LABELS[key];
+    });
     if (show_sqla_time_granularity) {
-      updatedColumns = {
-        ...updatedColumns,
-        [TIME_FILTER_MAP.time_grain_sqla]: form_data.time_grain_sqla,
-      };
+      configs.columns[TIME_FILTER_MAP.time_grain_sqla] =
+        form_data.time_grain_sqla;
     }
-
     if (show_sqla_time_column) {
-      updatedColumns = {
-        ...updatedColumns,
-        [TIME_FILTER_MAP.granularity_sqla]: form_data.granularity_sqla,
-      };
+      configs.columns[TIME_FILTER_MAP.granularity_sqla] =
+        form_data.granularity_sqla;
     }
-
     if (show_druid_time_granularity) {
-      updatedColumns = {
-        ...updatedColumns,
-        [TIME_FILTER_MAP.granularity]: form_data.granularity,
-      };
+      configs.columns[TIME_FILTER_MAP.granularity] = form_data.granularity;
     }
-
     if (show_druid_time_origin) {
-      updatedColumns = {
-        ...updatedColumns,
-        [TIME_FILTER_MAP.druid_time_origin]: form_data.druid_time_origin,
-      };
+      configs.columns[TIME_FILTER_MAP.druid_time_origin] =
+        form_data.druid_time_origin;
     }
+  }
+  return configs;
+}
 
-    configs = {
-      ...configs,
-      columns: updatedColumns,
-      labels: updatedLabels,
-    };
+export default function getFilterConfigsFromFormdata(
+  form_data = {},
+  filters = undefined,
+) {
+  const configs = form_data.table_filter
+    ? getFilterConfigsFromTableChart(form_data)
+    : getFilterConfigsFromFilterBox(form_data);
+
+  // if current chart has preselected filters, update it
+  if (filters) {
```

Comment:
this function `getFilterConfigsFromFormdata`, is used when dashboard mount, and when user apply a filter. By adding this code block, will user applied filter will always be overwritten by dashboard preselected filters?

Reply:
Hmm, I don’t think this is actually in use. It comes from my previous attempt with some larger refactoring, which didn’t work out in the end. Will clean this up.